File: Java/Exercicio3/MembrosEstaticosEAbstratos/src/Funcionario.java
import java.util.Objects;
import static java.lang.Integer.parseInt;
import java.util.regex.Pattern;
import java.util.regex.Matcher;

public abstract class Funcionario {
    private String nome;

    private String CPF;

    private double salarioBase;

    public String getNome() {
        return nome;
    }

    protected void setNome(String nome) {
        this.nome = nome;
    }

    public String getCPF() {
        return CPF;
    }

    protected void setCPF(String CPF) {
        this.CPF = CPF;
    }

    public double getSalarioBase() {
        return salarioBase;
    }

    protected void setSalarioBase(double salarioBase) {
        this.salarioBase = salarioBase;
    }

    static boolean verificaCPF(String strCPF){
        strCPF = strCPF.replaceAll("[.-]", "");
        int Soma = 0;
        int Resto;
        if (Objects.equals(strCPF, "00000000000")) return false;
        for (int i = 1; i <= 9; i++) {
            Soma = Soma + parseInt(strCPF.substring(i-1, i)) * (11 - i);
        }
        Resto = (Soma * 10) % 11;
        if ((Resto == 10) || (Resto == 11))  Resto = 0;
        if (Resto != parseInt(strCPF.substring(9, 10)) ) return false;
        Soma = 0;
        for (int i = 1; i <= 10; i++) Soma = Soma + parseInt(strCPF.substring(i-1, i)) * (12 - i);
        Resto = (Soma * 10) % 11;
        if ((Resto == 10) || (Resto == 11))  Resto = 0;
        if (Resto != parseInt(strCPF.substring(10, 11) ) ) return false;
        return true;
    }
    public abstract double calculaSalario();

    @Override
    public String toString(){
        return "O funcionario é: " + this.getNome() + " O CPF é: " + this.getCPF() + " O Salário é: " + this.calculaSalario() + " |\n";
    }
}

File: Python/Funcionario/Funcionario.py
from abc import ABC, abstractmethod
import re

class Funcionario(ABC):
    def __init__(self, nome, cpf, salario_base):
        self.nome = nome
        self.cpf = cpf
        self.salario_base = salario_base

    @abstractmethod
    def calculaSalario(self):
        pass

    @staticmethod
    def verificaCPF(strCPF):
            Soma = 0
            Resto = 0
            strCPF = re.sub(r"[.-]", "", strCPF)
            if len(strCPF) < 11:
                return False 
            if strCPF == "00000000000":
                return False
            for i in range(1, 10):
                Soma += int(strCPF[i-1]) * (11 - i)
            Resto = (Soma * 10) % 11
            if Resto == 10 or Resto == 11:
                Resto = 0
            if Resto != int(strCPF[9]):
                return False
            Soma = 0
            for i in range(1, 11):
                Soma += int(strCPF[i-1]) * (12 - i)
            Resto = (Soma * 10) % 11
            if Resto == 10 or Resto == 11:
                Resto = 0
            if Resto != int(strCPF[10]):
                return False
            return True